### DISCUSSION TITLE
Authentication from native mobile app

### DISCUSSION THREAD

**Discussion by alexgolod** [dexidp/dex#2767] [Q&A]
Sorry for possible stupid questions but I'm new to this..
I'm trying to come up with a way to implement authentication for API, which will be used from web and native mobile app. I was able to make it work with web using google connector, but when I get to authentication in mobile app I want to avoid opening browser to confirm authentication and use native feature instead (e.g. android allows to select one of connected google accounts)

So the question: Is there a way to do that with dex? 
As possible solution that came to me, is use native auth, get id_token from auth in google and feed it token to dex, so dex can issue an access token that i can use with my API, couldn't find a way to that tough.
Would be really great if someone can point me to right direction, or share their experience with solving such problem

**Comment by easy6666** [comment]
邮件收到，谢谢~

**Comment by sagikazarmark** [comment]
Hey @alexgolod 

Sorry, I don't think this is possible at the moment. Lot of mobile apps open a browser window though and I don't think it's that bad at all.

**Comment by easy6666** [comment]
邮件收到，谢谢~
